test/e2e_node/memory_eviction_test.go

```diff
@@ -84,60 +84,73 @@ var _ = framework.KubeDescribe("MemoryEviction [Slow] [Serial] [Disruptive]", fu
 				framework.ExpectNoError(beErr, fmt.Sprintf("getting pod %s", besteffort.Name))
 				bestPh := best.Status.Phase
 
-				glog.Infof("Pod phase: guaranteed: %v, burstable: %v, besteffort: %v", gteedPh, burstPh, bestPh)
+				glog.Infof("pod phase: guaranteed: %v, burstable: %v, besteffort: %v", gteedPh, burstPh, bestPh)
 
 				if bestPh == api.PodRunning {
-					Expect(burstPh).NotTo(Equal(api.PodFailed), "Burstable pod failed before best effort pod")
-					Expect(gteedPh).NotTo(Equal(api.PodFailed), "Guaranteed pod failed before best effort pod")
+					Expect(burstPh).NotTo(Equal(api.PodFailed), "burstable pod failed before best effort pod")
+					Expect(gteedPh).NotTo(Equal(api.PodFailed), "guaranteed pod failed before best effort pod")
 				} else if burstPh == api.PodRunning {
-					Expect(gteedPh).NotTo(Equal(api.PodFailed), "Guaranteed pod failed before burstable pod")
+					Expect(gteedPh).NotTo(Equal(api.PodFailed), "guaranteed pod failed before burstable pod")
 				}
 
-				// When both besteffort and burstable have been evicted, return true, else false
+				// When both besteffort and burstable have been evicted, the test has completed.
 				if bestPh == api.PodFailed && burstPh == api.PodFailed {
-					return true
+					return nil
 				}
-				return false
+				return fmt.Errorf("besteffort and burstable have not yet both been evicted.")
 
-			}, 60*time.Minute, 5*time.Second).Should(Equal(true))
+			}, 60*time.Minute, 5*time.Second).Should(BeNil())
 
-			// Wait for available memory to decrease to a reasonable level before ending the test.
-			// This prevents interference with tests that start immediately after this one.
-			Eventually(func() bool {
-				glog.Infof("Waiting for available memory to decrease to a reasonable level before ending the test.")
+			// Wait for the memory pressure condition to disappear from the node status before continuing.
+			Eventually(func() error {
```

Comment:
It might be useful to set `--eviction-pressure-transition-period=1m` in `e2e_service.go` since the default is `5m`.